test/gtest/common/mem_buffer.cc

```diff
@@ -276,54 +276,41 @@ void mem_buffer::copy_from(void *dst, const void *src, size_t length,
     copy_between(dst, src, length, UCS_MEMORY_TYPE_HOST, src_mem_type);
 }
 
-typedef enum {
-    useMemcpy,
-    useCudaMemcpy,
-    useHipMemcpy,
-    nothing
-} copy_type_t;
-
-inline copy_type_t preferrable_copy_type(ucs_memory_type_t mem_type)
-{
-    switch (mem_type) {
-    case UCS_MEMORY_TYPE_HOST:
-    case UCS_MEMORY_TYPE_ROCM_MANAGED:
-        return useMemcpy;
-    case UCS_MEMORY_TYPE_CUDA:
-    case UCS_MEMORY_TYPE_CUDA_MANAGED:
-        return useCudaMemcpy;
-    case UCS_MEMORY_TYPE_ROCM:
-        return useHipMemcpy;
-    default:
-        return nothing;
-    }
-}
-
-inline bool use_copy_type(ucs_memory_type_t dst_mem_type,
-                          ucs_memory_type_t src_mem_type,
-                          copy_type_t copy_type)
+/* check both mem types are in the given set */
+bool mem_buffer::check_mem_types(ucs_memory_type_t dst_mem_type,
+                                 ucs_memory_type_t src_mem_type,
+                                 const int mem_types)
 {
-    return (preferrable_copy_type(dst_mem_type) == copy_type ||
-            preferrable_copy_type(dst_mem_type) == useMemcpy) &&
-           (preferrable_copy_type(src_mem_type) == copy_type ||
-            preferrable_copy_type(src_mem_type) == useMemcpy);
+    return (UCS_BIT(dst_mem_type) & mem_types) &&
+           (UCS_BIT(src_mem_type) & mem_types);
 }
 
-
 void mem_buffer::copy_between(void *dst, const void *src, size_t length,
                               ucs_memory_type_t dst_mem_type,
                               ucs_memory_type_t src_mem_type)
 {
-    if (use_copy_type(dst_mem_type, src_mem_type, useMemcpy)) {
+    const int host_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST) |
+                               UCS_BIT(UCS_MEMORY_TYPE_ROCM_MANAGED);
```

Comment:
why is this used with host types and without `#if HAVE_RCOM`?

Reply:
use hipMemcpy to copy ROCM_MANAGED